src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -685,18 +691,53 @@ private BoundExpression BindTupleExpression(TupleExpressionSyntax node, Diagnost
                 Error(diagnostics, ErrorCode.ERR_TupleExplicitNamesOnAllMembersOrNone, node);
             }
 
-            ImmutableArray<TypeSymbol> elements = elementTypes.ToImmutableAndFree();
+            var elementNamesArray = elementNames == null ?
+                                default(ImmutableArray<string>) :
+                                elementNames.ToImmutableAndFree();
 
-            var type = new TupleTypeSymbol(
-                elements,
-                elementNames == null ?
-                    default(ImmutableArray<string>) :
-                    elementNames.ToImmutableAndFree(),
-                node,
-                this,
-                diagnostics);
+            TupleTypeSymbol tupleTypeOpt = null;
+            var elements = elementTypes.ToImmutableAndFree();
+
+            if (hasNaturalType)
+            {
+                tupleTypeOpt = new TupleTypeSymbol(elements, elementNamesArray, node, this, diagnostics);
+            }
+
+            return new BoundTupleLiteral(node, boundArguments.ToImmutableAndFree(), elementNamesArray, tupleTypeOpt, hasErrors);
```

Comment:
That is actually an error checked by CheckTupleMemberName below.
Item1, Item2 ... are reserved since you would not be just inadvertently renaming, you would be really naming two things by the same name. 

There is similarity with anonymous types here - you are not allowed to shadow what already exists in underlying type.